src/main/java/com/google/codeu/servlets/CountryServlet.java

```diff
@@ -71,7 +90,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     request.setAttribute("code", countryCode);
     request.setAttribute("name", countryData.getName());
     request.setAttribute("messages", messages);
+    request.setAttribute("countryCode", countryCode);
+    request.setAttribute("category", category);
     request.setAttribute("isUserLoggedIn", userService.isUserLoggedIn());
-    request.getRequestDispatcher("/WEB-INF/country.jsp").forward(request, response);
+    if(category != null) {
```

Comment:
May also want to check category.isEmpty(). Also, do this before setting attributes, I think, because a null category will crash here:  `   request.setAttribute("category", category);`

Reply:
Fixed!